app/javascript/hub/MainMenuComponent.js

```diff
@@ -1,8 +1,27 @@
 export default function MainMenuComponent() {
-    const mainMenu = document.querySelector('[data-component="MainMenuComponent"]')
+    const mainMenu = document.querySelector('[data-component="MainMenuComponent"]');
+    highlightSelectedPageNavigation(mainMenu);
+    toggleSidebarExpandCollapse(mainMenu);
+}
+
+function highlightSelectedPageNavigation(mainMenu) {
     mainMenu.querySelectorAll('.menu-item').forEach((item) => {
         if (item.href === window.location.href) {
             item.classList.toggle('selected');
         }
     });
 }
+
+function toggleSidebarExpandCollapse(mainMenu) {
+    mainMenu.querySelector('.toggle').addEventListener("click", () => {
+        const classes = mainMenu.classList;
+        classes.toggle("expanded");
```

Comment:
this is kind of nitty, but i feel like it would be simpler if we let one of the states be the "default" appearance of the component and then only toggle ONE class on or off